fvm/transactionContractFunctionInvocator.go

```diff
@@ -0,0 +1,79 @@
+package fvm
+
+import (
+	"fmt"
+
+	"github.com/opentracing/opentracing-go"
+	traceLog "github.com/opentracing/opentracing-go/log"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/onflow/cadence/runtime/sema"
+
+	"github.com/onflow/flow-go/fvm/programs"
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/module/trace"
+)
+
+type TransactionContractFunctionInvocator struct {
+	contractLocation common.AddressLocation
+	functionName     string
+	arguments        []interpreter.Value
+	argumentTypes    []sema.Type
+	logger           zerolog.Logger
+}
+
+func NewTransactionContractFunctionInvocator(
+	contractLocation common.AddressLocation,
+	functionName string,
+	arguments []interpreter.Value,
+	argumentTypes []sema.Type,
+	logger zerolog.Logger) *TransactionContractFunctionInvocator {
+	return &TransactionContractFunctionInvocator{
+		contractLocation: contractLocation,
+		functionName:     functionName,
+		arguments:        arguments,
+		argumentTypes:    argumentTypes,
+		logger:           logger,
+	}
+}
+
+func (i *TransactionContractFunctionInvocator) Invoke(vm *VirtualMachine, ctx *Context, proc *TransactionProcedure, sth *state.StateHolder, programs *programs.Programs) (cadence.Value, error) {
+	var span opentracing.Span
+	if ctx.Tracer != nil && proc.TraceSpan != nil {
+		span = ctx.Tracer.StartSpanFromParent(proc.TraceSpan, trace.FVMExecuteTransaction)
```

Comment:
I think it shouldn't be `FVMExecuteTransaction` should be some other constant showing invoke meta transaction is happening